app/api/event_tag_endpoint.py

```diff
@@ -1,26 +1,31 @@
 from tracardi.service.storage.drivers.elastic.tag import add_tags, remove_tags, load_all_tags
 from pydantic import BaseModel
 from typing import List
-from fastapi import APIRouter
+from fastapi import APIRouter, Depends, HTTPException
 from app.config import server
 from tracardi.domain.value_object.bulk_insert_result import BulkInsertResult
+from .auth.authentication import get_current_user
 
 
 class TagForm(BaseModel):
     event_type: str = ""
     tags: List[str] = []
 
 
-router = APIRouter()
+router = APIRouter(
+    dependencies=[Depends(get_current_user)]
+)
 
 
 @router.post("/event/tag/add", tags=["event"], include_in_schema=server.expose_gui_api, response_model=dict)
 async def add_event_tag_endpoint(tag_form: TagForm):
-    updated, result = await add_tags(
+    result = await add_tags(
```

Comment:
Change the name of add_tags to add and instead of 
add_tags. Then use the full path to driver.

```
from tracardi.service.storage.driver import storage
```

And instead of add_tags just do : 
```
storage.drivers.elastic.tag.add(...) 
```
This is a better way to show that this is an driver operation.

See how its done in profile endpoint:

```
@router.get("/profiles/refresh", tags=["profile"], include_in_schema=server.expose_gui_api)
async def refresh_profile():
    try:
        return await storage.driver.profile.refresh()
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))
```

Reply:
Do the same for remove, etc.